source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1490,6 +1599,12 @@ Once the client has taken the lock it must do no I/O::
 
         newTopologyDescription = client.topologyDescription.copy()
 
+        # Ignore this update if the current topologyVersion is greater than
+        # the new ServerDescription's.
+        if not shouldHandleServerDescription(td, server):
```

Comment:
nit: it's easy to overlook the `not` here which is a python keyword, perhaps we should change the method name to something like `shouldDiscardServerUpdate` which better matches what we use the method for.

Reply:
I renamed shouldHandleServerDescription to isStaleServerDescription and shouldHandleError to isStaleError.